dashboard/test/lib/email_reminder_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+require 'email_reminder'
+
+class EmailReminderTest < ActiveSupport::TestCase
+  setup_all do
+    @student = create :student
+    @student.update(child_account_compliance_state: 'not_g')
+    @request = ParentalPermissionRequest.create(user_id: @student.id, parent_email: 'foo-parent@code.org')
```

Comment:
Please create a factory for `ParentalPermissionRequest` in `dashboard/test/factories/factories.rb`;

Or I would recommend creating a new factory file for each model/resource:
```ruby
# dashboard/test/factories/parental_permission_requests.rb
FactoryBot.define do
  factory :parental_permission_request do
    user
    sequence(:parent_email) { |n| "parent-#{n}@code.org" }
  end
end
```

```suggestion
    @request = create(:parental_permission_request, user: @student, parent_email: 'foo-parent@code.org')
```

Reply:
Unfortunately convention here is to cram everything into the single factories.rb file. Ideally we break this file into individual files like your second recommendation, however that should be part of a bigger effort, so we can maintain uniformity.